docs/concepts/json_schema.md

```diff
@@ -572,6 +725,63 @@ print(m.model_fields)
 """
 ```
 
+### `WithJsonSchema` annotation
+
+??? api "API Documentation"
+    [`pydantic.json_schema.WithJsonSchema`][pydantic.json_schema.WithJsonSchema]<br>
+
+The [`WithJsonSchema`][pydantic.json_schema.WithJsonSchema] annotation can be used to override the generated (base)
+JSON schema for a given type without the need to implement `__get_pydantic_core_schema__` on the type itself.
+
+This provides a way to set a JSON schema for types that would otherwise raise errors when producing a JSON schema,
+such as `Callable`, or types that have an `is-instance` core schema.
```

Comment:
Maybe we can add a `??? tip "How to implement..."` and implement something simple for `Callable`?

Also, can we add some reference to `is-instance`? Looks like a new concept for users reading this?

Reply:
@Kludex,

I've added the link to the core API docs re the `is-instance` schema reference.

What kind of example did you have in mind for the `Callable` case? Shortly after this text we give an example with a `PlainValidator`...

Reply:
I'm happy already.